examples/all-clusters-app/all-clusters-common/include/water-heater-mode.h

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/util/config.h>
+#include <cstring>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace WaterHeaterMode {
+
+const uint8_t ModeOff    = 0;
+const uint8_t ModeManual = 1;
+const uint8_t ModeTimed  = 2;
+
+/// This is an application level delegate to handle WaterHeaterMode commands according to the specific business logic.
+class ExampleWaterHeaterModeDelegate : public ModeBase::Delegate
+{
+private:
+    using ModeTagStructType             = detail::Structs::ModeTagStruct::Type;
+    ModeTagStructType modeTagsOff[1]    = { { .value = to_underlying(ModeTag::kOff) } };
+    ModeTagStructType modeTagsManual[1] = { { .value = to_underlying(ModeTag::kManual) } };
+    ModeTagStructType modeTagsTimed[1]  = { { .value = to_underlying(ModeTag::kTimed) } };
+
+    const detail::Structs::ModeOptionStruct::Type kModeOptions[3] = {
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Off"),
```

Comment:
documentation of fromCharString says to use `_span` instead

Reply:
Again - We have followed the pattern used in other mode clusters (e.g. dishwasher-mode.h, microwave-oven-mode.h) they all use this format.

So we were trying to be consistent with them.

Reply:
I have made the changes you suggested Andrei.